src/main/java/bio/terra/stairway/StairwayHook.java

```diff
@@ -26,4 +26,8 @@ default HookAction stateTransition(FlightContext context) throws InterruptedExce
   default Optional<StepHook> stepFactory(FlightContext context) throws InterruptedException {
     return Optional.empty();
   }
+
+  default Optional<FlightHook> flightFactory(FlightContext context) throws InterruptedException {
```

Comment:
optional: could use some more comments on the StairwayHook interface about what it's used for, and any guarantees about when its functions will be run.

Reply:
I added documentation to StairwayHook class and tried to provide cardinality info.